common/redispool/redis_pool.go

```diff
@@ -93,31 +93,96 @@ type Resp struct {
+	// new connection is slow.
+	MaxIdle int `json:"maxIdle"`
+
+	// Amount of time after which client closes idle connections.
+	// Should be less than server's timeout.
+	// Default is 5 minutes. -1 disables idle timeout check.
+	IdleTimeout commontime.Duration `json:"idleTimeout"`
+
+	// ConnectTimeout for go-redis is Dial timeout for establishing new connections.
+	// Default is 5 seconds.
 	ConnectTimeout commontime.Duration `json:"connectTimeout"`
-	MsgTimeout     commontime.Duration `json:"msgTimeout"`
-	Concurrency    int                 `json:"concurrency"`
-	Compatible     bool                `json:"compatible"`
-	MaxRetry       int                 `json:"maxRetry"`
-	MinBatchCount  int                 `json:"minBatchCount"`
-	WaitTime       commontime.Duration `json:"waitTime"`
+
+	MsgTimeout    commontime.Duration `json:"msgTimeout"`
+	Concurrency   int                 `json:"concurrency"`
+	Compatible    bool                `json:"compatible"`
+	MaxRetry      int                 `json:"maxRetry"`
+	MinBatchCount int                 `json:"minBatchCount"`
+	WaitTime      commontime.Duration `json:"waitTime"`
+
+	// MaxRetries is Maximum number of retries before giving up.
+	// Default is 3 retries; -1 (not 0) disables retries.
+	MaxRetries int `json:"maxRetries"`
+
+	// DB is Database to be selected after connecting to the server.
+	DB int `json:"DB"`
+
+	// ReadTimeout for socket reads. If reached, commands will fail
+	// with a timeout instead of blocking. Use value -1 for no timeout and 0 for default.
+	// Default is 3 seconds.
+	ReadTimeout commontime.Duration `json:"readTimeout"`
+
+	// WriteTimeout for socket writes. If reached, commands will fail
+	// with a timeout instead of blocking.
+	// Default is ReadTimeout.
+	WriteTimeout commontime.Duration `json:"writeTimeout"`
+
+	// Maximum number of socket connections.
+	// Default is 10 connections per every available CPU as reported by runtime.GOMAXPROCS.
+	PoolSize int `json:"poolSize"`
+
+	// Amount of time client waits for connection if all connections
+	// are busy before returning an error.
+	// Default is ReadTimeout + 1 second.
+	PoolTimeout commontime.Duration `json:"poolTimeout"`
+
+	// Connection age at which client retires (closes) the connection.
+	// Default is to not close aged connections.
+	MaxConnAge commontime.Duration `json:"maxConnAge"`
+
+	// Use the specified Username to authenticate the current connection
+	// with one of the connections defined in the ACL list when connecting
+	// to a Redis 6.0 instance, or greater, that is using the Redis ACL system.
+	Username string `json:"username"`
+
+	// WithTLS whether open TLSConfig
+	// if WithTLS is true, you should call WithEnableWithTLS,and then TLSConfig is not should be nil
+	// In this case you should call WithTLSConfig func to set tlsConfig
+	WithTLS bool `json:"withTLS"`
+
+	// TLS Config to use. When set TLS will be negotiated.
+	TlsConfig *tls.Config
```

Comment:
both tlsConfig itself and fields in tls.Config have not `json` tag for unmarshalling from JSON config.